package com.example.onlyone.domain.user.dto.request;

import com.example.onlyone.domain.user.entity.Gender;
import jakarta.validation.constraints.*;

import java.time.LocalDate;
import java.util.List;

public record ProfileUpdateRequestDto(
        @NotBlank(message = "닉네임은 필수입니다.")
        @Size(min = 2, max = 10, message = "닉네임은 2자 이상 10자 이하여야 합니다.")
        @Pattern(regexp = "^[가-힣a-zA-Z0-9]+$", message = "닉네임은 한글, 영문, 숫자만 사용 가능합니다.")
        String nickname,

        @NotNull(message = "생년월일은 필수입니다.")
        @Past(message = "생년월일은 과거 날짜여야 합니다.")
        LocalDate birth,

        String profileImage,

        @NotNull(message = "성별은 필수입니다.")
        Gender gender,

        @NotBlank(message = "시/도는 필수입니다.")
        String city,

        @NotBlank(message = "구/군은 필수입니다.")
        String district,

        @NotNull(message = "관심사는 필수입니다.")
        List<String> interestsList
) {
}
